dart/lib/src/sentry_client.dart

```diff
@@ -101,20 +101,16 @@ class SentryClient {
       platform: event.platform ?? sdkPlatform,
     );
 
-    if (stackTrace != null &&
-        event.throwable == null &&
-        event.exception == null) {
-      final frames = _stackTraceFactory.getStackFrames(stackTrace);
+    if (event.exception != null) return event;
 
-      if (frames != null && frames.isNotEmpty) {
-        event = event.copyWith(stackTrace: SentryStackTrace(frames: frames));
-      }
-    } else if (event.throwable != null && event.exception == null) {
+    if (event.throwable != null) {
       final sentryException = _exceptionFactory
           .getSentryException(event.throwable, stackTrace: stackTrace);
 
-      event = event.copyWith(exception: sentryException);
-    } else if (_options.attachStackTrace && event.exception == null) {
+      return event.copyWith(exception: sentryException);
+    }
+
+    if (stackTrace != null || _options.attachStackTrace) {
       stackTrace ??= StackTrace.current;
```

Comment:
this condition seems wrong.

`stackTrace` can be null, and if null, get the StackTrace.current only if attachStackTrace is enabled

Reply:
sorry I don't understand, here :
- if stacktrace is null AND _options.attachStackTrace is true => we use StackTrace.current
- if stacktrace is not null AND _options.attachStackTrace is false =>  we use stacktrace
- if stacktrace is not null AND _options.attachStackTrace is true =>  we use stacktrace
which case is wrong ?

Reply:
actually its right, I read as `&&` instead of `||`, my bad sorry.
lets just be sure that we cover all the cases via unit tests

Reply:
✅ added few tests to be sure